test/nextcloud/1.1.0/templates/backup-postgres-hook.yaml

```diff
@@ -0,0 +1,57 @@
+{{- if .Values.ixChartContext.isUpgrade -}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "pre-upgrade-hook2"
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded
+    rollme: {{ randAlphaNum 5 | quote }}
+spec:
+  template:
+    metadata:
+      name: "pre-upgrade-hook2"
+    spec:
+      restartPolicy: Never
+      serviceAccountName: "{{ template "nextcloud.serviceAccountName" . }}"
+      containers:
+      - name: {{ .Chart.Name }}-postgres-backup
+        image: "postgres:13.1"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        env:
+        - name: POSTGRES_USER
+          valueFrom:
+            secretKeyRef:
+              name: db-details
+              key: db-user
+        - name: POSTGRES_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: db-details
+              key: db-password
+        - name: BACKUP_NAME
+          value: {{ template "postgres.backupName" . }}
+        volumeMounts:
+          - name: postgres-data
+            mountPath: /var/lib/postgresql/data
```

Comment:
It's been a minute since I've messed with docker containers, but this is the host path for which the data is actually stored, right? Not inside the container?

Reply:
Nope, this is the path inside the container.

Reply:
Okay thanks, do we have a method for maintaining the copies of the databases? i.e . are we only keeping last 2 upgrades worth of backups or is that already being managed with this commit?

Reply:
So in other words would we end up with a long list of postgres backups and if we are automatically trimming it down ?

Reply:
Correct, are we managing the copies of the backups that we take.

Reply:
For now we are not. Backup is only going to happen on chart upgrade which i anticipate is not going to be very often.
But we can make keeping the backup optional when we do restore. So down the road i intend to restore the backup after upgrade of chart completes and then if user wants to store backup of database, we can keep it there, otherwise we can delete the backup file.